File: android/app/src/main/java/org/evergreenlabs/mytwitter/data/PublicConfig.kt
package org.evergreenlabs.mytwitter.data

import com.google.firebase.firestore.Exclude
import com.google.firebase.firestore.IgnoreExtraProperties
import java.util.Date

@IgnoreExtraProperties
data class UsageStats(
    val postsRead: Int? = null,
    val postsReadCumulative: Int? = null,
    val cyclePostsRead: Int? = null,
    val priorCyclesPostsRead: Int? = null,
    val todayPostsRead: Int? = null,
    val todayDate: String? = null,
    val projectCap: Int? = null,
    val capResetDay: Int? = null,
    val pricePerPostUsd: Double? = null,
    val estimatedCostUsd: Double? = null,
    val updatedAt: Date? = null,
)

@IgnoreExtraProperties
data class PublicConfig(
    val invitesEnabled: Boolean? = null,
    val lastRefreshedAt: Date? = null,
    val usage: UsageStats? = null,
) {
    // Must Exclude — Firestore treats Boolean `invitesEnabled` + `isInvitesEnabled` as conflicts.
    @get:Exclude
    val isInvitesEnabled: Boolean get() = invitesEnabled == true
}

File: ios/MyTwitter/Models/PublicConfig.swift
import Foundation
import FirebaseFirestore

struct UsageStats: Codable, Hashable {
    var postsRead: Int?
    var postsReadCumulative: Int?
    var cyclePostsRead: Int?
    var priorCyclesPostsRead: Int?
    var todayPostsRead: Int?
    var todayDate: String?
    var projectCap: Int?
    var capResetDay: Int?
    var pricePerPostUsd: Double?
    var estimatedCostUsd: Double?
    var updatedAt: Date?
}

struct PublicConfig: Codable, Hashable {
    var invitesEnabled: Bool?
    var lastRefreshedAt: Date?
    var usage: UsageStats?

    var isInvitesEnabled: Bool { invitesEnabled == true }
}
